### PR TITLE
Integation test additional media 2

### PR DESCRIPTION
**Pull request by NadiaKishchuk**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: Streetcode/Streetcode.WebApi/appsettings.IntegrationTests.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
   "ConnectionStrings": {
-    "DefaultConnection": "Server=(localdb)\\MSSQLLocalDB;Database=_Change_Me;Trusted_Connection=True;MultipleActiveResultSets=true"
```
Comment (Katerix) [inline Streetcode/Streetcode.WebApi/appsettings.IntegrationTests.json:3]: Did you mean to commit this?

File: Streetcode/Streetcode.WebApi/appsettings.Local.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
   "ConnectionStrings": {
-    "DefaultConnection": "Server=localhost;Database=StreetcodeDb;User Id=sa;Password=Admin@1234;MultipleActiveResultSets=true"
```
Comment (Katerix) [inline Streetcode/Streetcode.WebApi/appsettings.Local.json:3]: same

File: Streetcode/Streetcode.XIntegrationTest/appsettings.Local.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+  "ConnectionStrings": {
+    "DefaultConnection": "Server=DESKTOP-I7Q35NQ\\SQLEXPRESS;Database=_TestStreetcodeDb;User Id=sa;Password=Admin@1234;MultipleActiveResultSets=true"
```
Comment (Katerix) [inline Streetcode/Streetcode.XIntegrationTest/appsettings.Local.json:3]: Do you need to commit this?

File: Streetcode/Streetcode.XIntegrationTest/appsettings.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+  "ConnectionStrings": {
+    "DefaultConnection": "Server=DESKTOP-I7Q35NQ\\SQLEXPRESS;Database=StreetcodeDb;User Id=sa;Password=Admin@1234;MultipleActiveResultSets=true"
```
Comment (Katerix) [inline Streetcode/Streetcode.XIntegrationTest/appsettings.json:3]: same here

**Review by Katerix — COMMENTED** [review]
You have done such a big job! Pretty clean and readable code. You have my questions added to some files. One last thing is an optional advice to remove useless spaces between curly braces { }.
Looking forward to getting  your response.

File: Streetcode/Streetcode.XIntegrationTest/ControllerTests/AdditionalContent/CoordinateControllerTests.cs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        {
+        }
+
+        [Fact]
+        public async Task GetByStreetcodeId_SuccsessStatusCode()
+        {
+            int id = 1;
+            var response = await client.GetByIdAsync(id);
+
+            Assert.True(response.IsSuccessStatusCode);
```
Comment (LanchevychMaxym) [inline Streetcode/Streetcode.XIntegrationTest/ControllerTests/AdditionalContent/CoordinateControllerTests.cs:23]: In addition to status code you should check if returned object is the one we expect

File: Streetcode/Streetcode.XIntegrationTest/ControllerTests/AdditionalContent/CoordinateControllerTests.cs [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+            if (response.IsSuccessStatusCode)
+            {
+                var returnedValue = CaseIsensitiveJsonDeserializer.Deserialize<IEnumerable<StreetcodeCoordinateDTO>>(response.Content);
+
+                Assert.NotNull(returnedValue);
+
+                if (returnedValue.Count() != 0)
+                {
+                    Assert.Equal(id, returnedValue.FirstOrDefault(c => c.StreetcodeId == id)?.StreetcodeId);
+                }
```
Comment (LanchevychMaxym) [inline Streetcode/Streetcode.XIntegrationTest/ControllerTests/AdditionalContent/CoordinateControllerTests.cs:41]: Any logic in tests is bad practice

File: Streetcode/Streetcode.XIntegrationTest/ControllerTests/AdditionalContent/CoordinateControllerTests.cs [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+            }
+        }
+
+        [Fact]
+        public async Task GetByStreetcodeId_DoesntExist_ReturnBadRequest()
+        {
+            var response = await client.GetByStreetcodeId(-10);
+            Assert.False(response.IsSuccessStatusCode);
```
Comment (LanchevychMaxym) [inline Streetcode/Streetcode.XIntegrationTest/ControllerTests/AdditionalContent/CoordinateControllerTests.cs:49]: the same with response, we need to add more asserts

File: Streetcode/Streetcode.XIntegrationTest/ControllerTests/AdditionalContent/SubtitleControllerTests.cs [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+        [Fact]
+        public async Task GetAll_ReturnSuccessStatusCode()
+        {
+            var response = await this.client.GetAllAsync();
+            var returnedValue = CaseIsensitiveJsonDeserializer.Deserialize<IEnumerable<SubtitleDTO>>(response.Content);
+
+            Assert.Multiple(
+                () => Assert.True(response.IsSuccessStatusCode),
+                () => Assert.NotNull(returnedValue),
+                () => Assert.Equal(10, returnedValue.Count()));
```
Comment (LanchevychMaxym) [inline Streetcode/Streetcode.XIntegrationTest/ControllerTests/AdditionalContent/SubtitleControllerTests.cs:23]: This 10 should be a variable

File: Streetcode/Streetcode.XIntegrationTest/ControllerTests/AdditionalContent/SubtitleControllerTests.cs [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+            Assert.Multiple(
+                () => Assert.Equal(id, returnedValue?.Id),
+                () => Assert.True(response.IsSuccessStatusCode));
+        }
+
+        [Fact]
+        public async Task GetById_Incorrect_ReturnBadRequest()
+        {
+            var response = await this.client.GetByIdAsync(-100);
```
Comment (LanchevychMaxym) [inline Streetcode/Streetcode.XIntegrationTest/ControllerTests/AdditionalContent/SubtitleControllerTests.cs:54]: -100 should be a var

File: Streetcode/Streetcode.XIntegrationTest/ControllerTests/AdditionalContent/TagControllerTests.cs [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+                () => Assert.Equal(System.Net.HttpStatusCode.BadRequest, response.StatusCode),
+                () => Assert.False(response.IsSuccessStatusCode));
+        }
+
+        //return only with the equaltitle
+        [Theory]
+        [InlineData("writer")]
+        public async Task GetByTitle_ReturnSuccessStatusCode(string title)
+        {
+            var response = await client.GetResponse($"/GetTagByTitle/{title}");
```
Comment (LanchevychMaxym) [inline Streetcode/Streetcode.XIntegrationTest/ControllerTests/AdditionalContent/TagControllerTests.cs:79]: Why in one part of code we use  base methods but here we have something else

File: Streetcode/Streetcode.XIntegrationTest/ControllerTests/Media/Images/ArtControllerTests.cs [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+                () => Assert.Equal(System.Net.HttpStatusCode.BadRequest, response.StatusCode),
+                () => Assert.False(response.IsSuccessStatusCode));
+        }
+
+        [Theory]
+        [InlineData(1)]
+        public async Task GetByStreetcodeId_ReturnSuccessStatusCode(int streetcodeId)
+        {
+            var response = await client.GetByStreetcodeId(streetcodeId);
+            var returnedValue = CaseIsensitiveJsonDeserializer.Deserialize<IEnumerable< ArtDTO>>(response.Content);
```
Comment (LanchevychMaxym) [inline Streetcode/Streetcode.XIntegrationTest/ControllerTests/Media/Images/ArtControllerTests.cs:54]:
```suggestion
            var returnedValue = CaseIsensitiveJsonDeserializer.Deserialize<IEnumerable<ArtDTO>>(response.Content);
```

File: Streetcode/Streetcode.XIntegrationTest/ControllerTests/Utils/StreetcodeClient.cs [hunk 12]
```diff
@@ -0,0 +1,47 @@
+﻿namespace Streetcode.XIntegrationTest.ControllerTests.Utils
+{
+    using RestSharp;
+    using RestSharp.Serializers;
+    using System.Threading.Tasks;
+
+    public class StreetcodeClient
+    {
+        public RestClient Client { get; }
+        public HttpClient cli;
```
Comment (LanchevychMaxym) [inline Streetcode/Streetcode.XIntegrationTest/ControllerTests/Utils/StreetcodeClient.cs:10]: If we have RestClient we don't need HttpClient

Comment (LanchevychMaxym) [reply Streetcode/Streetcode.XIntegrationTest/ControllerTests/Utils/StreetcodeClient.cs:10]: RestClient can be as a field

File: Streetcode/Streetcode.XIntegrationTest/appsettings.Local.json (hunk 3 above)
Comment (LanchevychMaxym) [reply Streetcode/Streetcode.XIntegrationTest/appsettings.Local.json:3]: We need this string, but the one that is used to connect to local docker db

**Review by Tysyatsky — APPROVED** [review]
Great work! It needs more appreciation.

**Pull request closed without merging**